rest_framework_json_api/relations.py

```diff
@@ -116,7 +116,11 @@ def get_links(self, obj=None, lookup_field='pk'):
         })
         self_link = self.get_url('self', self.self_link_view_name, self_kwargs, request)
 
-        related_kwargs = {self.related_link_url_kwarg: kwargs[self.related_link_lookup_field]}
+        if self.related_link_url_kwarg == 'pk':
```

Comment:
You have more insight into generation of this links. Can you briefly explain why this is needed? In what case would this if match?

Reply:
Oh yeah!
The `if` would match in any case you didn't pass the argument when declaring a field, or the argument value equals 'pk'.

When `related_link_url_kwarg == 'pk'` we can suppose the url was declared like this:
```python
url(r'^authors/(?P<pk>[^/.]+)/(?P<related_field>\w+)/$',
        AuthorViewSet.as_view({'get': 'retrieve_related'}),
        name='author-related'),
```
and the route has 2 parameters: `pk` and `related_field`. Exactly like `self_link`. That is why we do: `related_kwargs = self_kwargs` in the next line.

In any other cases (`related_link_url_kwarg != 'pk'`) we assume that we have "old-style" urls declaration like this:
```python
url(r'^authors/(?P<author_pk>[^/.]+)/bio/$',
        AuthorBioViewSet.as_view({'get': 'retrieve'}),
        name='author-bio'),
```

Reply:
makes sense. What do you think about adding a small doc string explaining this?

Also do you think this could break backwards compatibility in case someone used `pk` in its related link? Although documentation states differently I guess.

Reply:
Your added docstring looks good. Thanks.